### PR TITLE
Adds entries to Makefile and RELEASING.md to automate canonical updates

### PR DESCRIPTION
**Pull request by emanic**
## Description

- Adds one recipe to Makefile and instructions to RELEASING.md to automate the updates of the canonical_url tags 

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by ozdanborne** [comment]
I ran `make canonical_urls` and am seeing some weird changes, like:

```
+canonical_url: https://docs.projectcalico.org/./master/getting-started/bare-metal/bare-metal
 canonical_url: 'https://docs.projectcalico.org/v3.0/getting-started/bare-metal/bare-metal'
```

and 
```
 ---
 title: Project Calico Documentation
+canonical_url: https://docs.projectcalico.org/./.html
 description: Home
 layout: docwithnav
```

**Comment by emanic** [comment]
Uh oh, let me look

**Comment by emanic** [comment]
@ozdanborne I couldn't reproduce this....did you pass in the value for version?

**Comment by ozdanborne** [comment]
Ah I did not. Can you add a quick note on that in the Makefile as well?

**Comment by emanic** [comment]
OK good idea

File: RELEASING.md [hunk 1]
```diff
@@ -76,6 +76,8 @@ release is announced.
 1. Push the tag.
 1. Create a GitHub release on the Calico repo, upload the release file (calico_node/release-<VERSION>.tgz), and post the updated
    release notes (see above for details).
+1. Run `make add_canonical_urls VERSION=vX.Y` to add canonical URLs to the metadata of the latest release files. Pass in the number of the new latest release via `VERSION`. Example: `make add_canonical_urls VERSION=v1.3`, where `1.3` is the new latest release.
```
Comment (ozdanborne) [inline RELEASING.md]: the use of the word `latest` here is misleading. I think you mean the release that was previously the latest but is now about to become previous. right?  (I'm not suggesting any chnage yet, just trying to make sure I understand)

File: RELEASING.md [hunk 2]
```diff
@@ -76,6 +76,9 @@ release is announced.
 1. Push the tag.
 1. Create a GitHub release on the Calico repo, upload the release file (calico_node/release-<VERSION>.tgz), and post the updated
    release notes (see above for details).
+1. Run `make update_canonical_urls OLD=vX.Y NEW=vX.Y` to switch the canonical URLs to the latest release version number. Pass in the number of the previous release via `OLD` and the number of the current latest release via `NEW`. Example: `make update_canonical_urls OLD=v3.0 NEW=v3.1`, where `3.0` was the previous latest and `3.1` is the new latest release.
+1. Test the changes locally then open a pull request, make sure it passes CI and get it reviewed.
```
Comment (ozdanborne) [inline RELEASING.md]: Can we be slightly more explicit by saying "Test the changes locally with `make htmlproofer`"?

**Pull request merged**